crates/node/src/roles/worker.rs

```diff
@@ -104,6 +112,8 @@ impl WorkerRole {
     }
 
     pub async fn start(self) -> anyhow::Result<()> {
+        Self::provision_worker_role(self.metadata_writer).await?;
```

Comment:
I wonder if we should delay the transition to Active to happen later after we successfully start the PPM to ensure that we are ready to be scheduled as a worker, rather than activating ourselves early and later failing during startup.

Reply:
On the path from here to running the PPM are `TaskCenter::spawns` that can fail in case of a shut down. I can let the PPM update the value to harden things in case we introduce other failing operations in the future.